packages/components/Badge/src/Badge.tsx

```diff
@@ -37,17 +39,28 @@ export const Badge = compose<BadgeType>({
     const Slots = useSlots(userProps, (layer) => badgeLookup(layer, userProps));
 
     return (final: BadgeProps, ...children: ReactNode[]) => {
-      const { icon, iconPosition, size, ...mergedProps } = mergeProps(badge, final);
+      const { icon, iconPosition, size, shadow, ...mergedProps } = mergeProps(badge, final);
       const showContent = size !== 'tiny' && size !== 'extraSmall';
       const showIcon = size !== 'tiny';
+      const BadgeComponent = shadow ? Slots.shadow : Fragment;
```

Comment:
I still think you don't need to do this, if there is no shadow token set, the shadow slot will only render it's children. Is there an issue removing the use of fragment?

Reply:
Yes we basically have the same code already in the Shadow component - if the shadowToken is undefined we just return a fragment around the children. Think you just need 

<Slots.shadow>
   <Slots.root ...>
       ...
   </Slots.root>
</Slots.shadow>

Reply:
Ok, I'll remove it